### PR TITLE
runtime: ensure containers get streams closed on exit

### PR DESCRIPTION
**Pull request by smola**

### CODE REVIEW

**Comment by smola** [comment]
Depends on https://github.com/bblfsh/server/pull/10

File: runtime/container.go [hunk 1]
```diff
@@ -37,35 +39,57 @@ type Command interface {
...
+	defer func() {
+		derr = c.Container.Destroy()
+		streams := []interface{}{
+			c.process.Stdin,
+			c.process.Stdout,
+			c.process.Stderr,
+		}
+		for _, s := range streams {
+			if c, ok := s.(io.Closer); ok {
+				if err := c.Close(); err != nil && derr != nil {
```
Comment (abeaumont) [inline runtime/container.go]: I've not enough background to judge this code, but it looks weird this `derr != nil` check, why is it checked?

Comment (smola) [inline runtime/container.go]: If there are multiple errors, only the first one is returned. So it does not override the error if it is already set.

Comment (abeaumont) [inline runtime/container.go]: then it should be `derr == nil`, otherwise you're overriding it, aren't you?

**Pull request merged**